### ISSUE TITLE
Suggestions and Improvements for the Analysis File

### ISSUE THREAD

**Issue by HrayrMuradyan** [UBC-MDS/airline-customer-satisfaction-predictor#2]
Overall a good job on the analysis file. I want to add some comments/suggestions and I am opening an issue as the PR is already closed. Please go over the suggestions and let me know what you think:

1. **Summary**: 

"... The goal is to identify which attributes (such as flight-related, service quality, and demographics) contribute most significantly to passenger satisfaction...." - I don't think we are trying to find the attributes that contribute most to customer satisfaction (at least it would be hard with k-NN), we rather try to come up with a predictive model that successfully predicts customer satisfaction. For finding the significant features we should use models like Logistic Regression, Trees or other models that can provide feature importance.

2. **Evaluation Metric**: 

F2-Score is fine, we can also use accuracy as the classes are approximately balanced.

3. **EDA**:

- ID and "Unnamed: 0" are unique identifiers and no need to include them in the correlation matrix.
- Only one column has missing values and it's numeric. No need to feel categorical variables with most frequent one.

4. **Feature Engineering**:
- We haven't encoded age into age bins and same for flight distance. I suggest leaving them as they are, they make sense as numeric variables.

**Comment by shengjiayuu** [comment]
I completely agree with your suggestion. Will fix it later.

**Comment by AzinPiran** [comment]
I agree with  your suggestion, the EDA should be again, we have some problems that should be fixed, maybe we can revert the pull request or create another branch. 
The things should be reviwed again:
1- The imputer is not needed for all of the columns
2- The correlation do not need to done for all of the columns
3- Maybe we can ignore the binary encoder for the target value. But this should be again checked in the process of modeling.
4- Maybe we just need to have one pipeline.

**Comment by AzinPiran** [comment]
All of the issues have been fixed based on pull request #12.
